updatersync/src/main/java/nl/openweb/hippo/groovy/watch/FileSystemWatcher.java

```diff
@@ -290,37 +298,45 @@ void start() {
 
         void processChange(final WatchEvent.Kind<?> kind, final Path changedAbsPath, final boolean isDirectory) {
             if (isDirectory) {
-                if (kind == StandardWatchEventKinds.ENTRY_CREATE) {
+                processDirectoryChange(kind, changedAbsPath);
+            } else {
+                processFileChange(kind, changedAbsPath);
+            }
+        }
+
+        private void processDirectoryChange(final WatchEvent.Kind<?> kind, final Path changedAbsPath) {
+            if (kind == StandardWatchEventKinds.ENTRY_CREATE) {
```

Comment:
Kind of syntax flavor / micro optimalisation, This could be rewitten to a switch statement, the bytecodes get generated from it is more optimized

Reply:
Not really a good idea, A switch works with the byte, short, char, and int primitive data types, some wrappers, and Enum entries. This doesn't apply to this check.

Reply:
Made the assumption it was a enum the way it was written